app/controllers/administrateurs/procedures_controller.rb

```diff
@@ -194,21 +194,56 @@ def clone
     def archive
       procedure = current_administrateur.procedures.find(params[:procedure_id])
 
-      if params[:new_procedure].present?
-        new_procedure = current_administrateur.procedures.find(params[:new_procedure])
-        procedure.update!(replaced_by_procedure_id: new_procedure.id)
+      if procedure.update(closing_params)
+        procedure.close!
+        if (procedure.dossiers.not_archived.state_brouillon.present? || procedure.dossiers.not_archived.state_en_construction_ou_instruction.present?) && procedure.closing_reason == Procedure.closing_reasons.fetch('other')
+          redirect_to admin_procedure_closing_notification_path
+        else
+          flash.notice = "Démarche close"
+          redirect_to admin_procedure_path(id: procedure.id)
+        end
+      else
+        flash.alert = procedure.errors.full_messages
+        redirect_to admin_procedure_close_path
       end
 
-      procedure.close!
-
-      flash.notice = "Démarche close"
-      redirect_to admin_procedures_path
-
     rescue ActiveRecord::RecordNotFound
       flash.alert = 'Démarche inexistante'
       redirect_to admin_procedures_path
     end
 
+    def closing_notification
+      @procedure = current_administrateur.procedures.find(params[:procedure_id])
+      @users_brouillon_count = @procedure.dossiers.not_archived.state_brouillon.map(&:user).uniq.count
```

Comment:
on peut faire ca pour éviter potentiellement de charger des milliers d'user

```suggestion
      @users_brouillon_count = @procedure.dossiers.not_archived.state_brouillon.count('distinct user_id')
```

Reply:
:+1: